zvariant/src/display.rs

```diff
@@ -0,0 +1,601 @@
+            }
+            write!(f, "0x{:02x}", num)
+        }
+        Value::Bool(boolean) => {
+            write!(f, "{}", boolean)
+        }
+        Value::I16(num) => {
+            if type_annotate {
+                f.write_str("int16 ")?;
+            }
+            write!(f, "{}", num)
+        }
+        Value::U16(num) => {
+            if type_annotate {
+                f.write_str("uint16 ")?;
+            }
+            write!(f, "{}", num)
+        }
+        Value::I32(num) => {
+            // Never annotate this type because it is the default for numbers
+            write!(f, "{}", num)
+        }
+        Value::U32(num) => {
+            if type_annotate {
+                f.write_str("uint32 ")?;
+            }
+            write!(f, "{}", num)
+        }
+        Value::I64(num) => {
+            if type_annotate {
+                f.write_str("int64 ")?;
+            }
+            write!(f, "{}", num)
+        }
+        Value::U64(num) => {
+            if type_annotate {
+                f.write_str("uint64 ")?;
+            }
+            write!(f, "{}", num)
+        }
+        Value::F64(num) => {
+            if num.fract() == 0. {
+                // Add a dot to make it clear that this is a float
+                write!(f, "{}.", num)
+            } else {
+                write!(f, "{}", num)
+            }
+        }
+        Value::Str(string) => {
+            if string.find('\'').is_some() {
+                write!(f, "\"{}\"", string)
+            } else {
+                write!(f, "'{}'", string)
+            }
+        }
+        Value::Signature(val) => {
+            if type_annotate {
+                f.write_str("signature ")?;
+            }
+            write!(f, "'{}'", val)
+        }
+        Value::ObjectPath(val) => {
+            if type_annotate {
+                f.write_str("objectpath ")?;
+            }
+            write!(f, "'{}'", val)
+        }
+        Value::Value(child) => {
+            f.write_char('<')?;
+
+            // Always annotate types in nested variants, because they
+            // are (by nature) of varible type.
+            value_display_fmt(child, f, true)?;
+
+            f.write_char('>')?;
+            Ok(())
+        }
+        Value::Array(array) => array_display_fmt(array, f, type_annotate),
+        Value::Dict(dict) => dict_display_fmt(dict, f, type_annotate),
+        Value::Structure(structure) => structure_display_fmt(structure, f, type_annotate),
+        #[cfg(feature = "gvariant")]
+        Value::Maybe(maybe) => maybe_display_fmt(maybe, f, type_annotate),
+        #[cfg(unix)]
+        Value::Fd(handle) => {
+            if type_annotate {
+                f.write_str("handle ")?;
+            }
+            write!(f, "{}", handle)
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::HashMap, convert::TryFrom};
+
+    use crate::{Fd, ObjectPath, Signature};
+
+    use super::*;
+
+    #[test]
+    fn u8_display() {
```

Comment:
Really appreciate you adding tests. However, I don't think, the display repr is important enough to warrant so many tests and adding to CI. Just a handul (10-15) random simple and complex cases would be more than enough IMO.